### ISSUE TITLE
Can't use asar with yarn workspaces and hoisted deps anymore in Windows

### ISSUE THREAD

**Issue by jlongster** [electron-userland/electron-builder#2962]
* **Version**: 20.15.3, which is on "next" and I think represents master right now
* **Target**: Windows

The PR https://github.com/electron-userland/electron-builder/pull/2948 tried to fix missing files when the electron app is a sub-package and yarn workspaces hoisted modules into the root dir. That PR still works for me, but the PR was closed in favor of this commit: https://github.com/electron-userland/electron-builder/commit/c31f6a1e6dd83cbf7170b89cf228f73c00a650ca

I don't really understand why there are so many changes in that commit. Many of them seem unrelated. But the current code only works if `asar` is disabled. When it's disabled, it correctly builds with the right files, but when enabled I get the following error:

<img width="893" alt="screen shot 2018-05-29 at 11 23 05 am" src="https://user-images.githubusercontent.com/17031/40668755-4dc70bd2-6333-11e8-903f-a3014d6f65db.png">

**Comment by lukeautry** [comment]
I think I'm seeing a similar/related thing using lerna w/ yarn workspaces.

Got slightly further by adding the electron app's workspace package to nohoist, but even then, node_modules in the asar is missing subdependencies of sibling packages.

**Comment by jlongster** [comment]
I think that's because as far as I can tell 20.15.3 is the only version that has https://github.com/electron-userland/electron-builder/pull/2948. So before that version, you're seeing the normal missing file. 20.15.3 has the fix, but at least on Windows with asar, produces the above error for me. Are you using asar?

**Comment by jlongster** [comment]
It does indeed work on macOS for me. So it's only a Windows issue.

**Comment by bhj** [comment]
@jlongster Same error/issue here and I'm using npm, so that PR seems to have unintended side-effects (might want to update title with error msg?)

On `20.15.3` Windows ASAR build gives `Error: File "" not under the source directory "C:\Users\Me\project\node_modules"`. MacOS works. Both work on previous versions. Pretty simple config:

```
{
  "files": [
    "constants/**",
    "server/**",
    "project.config.js",
    // use webpack output path but exclude subfolders
    "dist/*",
  ],
  "mac": {
    "target": "dmg",
  },
  "win": {
    "target": "nsis",
  },
}

```

**Comment by jlongster** [comment]
@develar See ^ if you haven't, I'd try to help debug but the PR that introduced this has so many changes for some reason. Did you mean to include all those changes in the PR?

**Comment by fubhy** [comment]
I am also seeing a new issue with the merged changes: #2987

**Comment by jlongster** [comment]
@develar So to keep pinging you, but I'm happy to help find a fix, it's just really hard to do it on my own because the changeset on the commit is so big. Any chance you could steer me in the right direction?

This will be the last time I ping you - otherwise I guess I'll try to figure it out on my own.

**Comment by timfish** [comment]
I've spent a while trying to reproduce this in the simplest possible form. The hoisted build only breaks when `asarUnpack` is used. Example repo to follow...

**Comment by timfish** [comment]
This is reproducible with `electron-builder@20.25.0`

Use the example repository: https://github.com/timfish/builder-workspace-demo

```
> yarn
> yarn release
```

Remove `asarUnpack` from `config.yml`, and the build completes successfully!

**Comment by gemma-ferreras** [comment]
It is happening to me in mac. If I remove the option

` "asarUnpack": [
      "plugins/"
    ],`

Then, I am able to make the package, with it, it is not possible right now. Using electron-builder 20.26.0

**Comment by timfish** [comment]
This looks like a duplicate of #3223 which has been fixed in 20.28.2. 

Can this be closed now?
